tpm/src/ioctl.rs

```diff
@@ -0,0 +1,434 @@
+// Copyright © 2022, Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::anyhow;
+use byteorder::{BigEndian, ReadBytesExt};
+use std::convert::TryInto;
+use thiserror::Error;
+
+pub const TPM_CRB_BUFFER_MAX: usize = 3968; // 0x1_000 - 0x80
+pub const TPM_SUCCESS: u32 = 0x0;
+
+/*
+ * Structures required to process Request and Responses of Control commands
+ * used by control channel over UNIX socket interface
+ *
+ * All messages contain big-endian data.
+ *
+ * Reference: https://github.com/stefanberger/swtpm/blob/master/man/man3/swtpm_ioctls.pod
+ */
+#[derive(Debug, Clone, Copy)]
+pub enum Commands {
+    CmdGetCapability = 1,
+    CmdInit,                // 2
+    CmdShutdown,            // 3
+    CmdGetTpmEstablished,   // 4
+    CmdSetLocality,         // 5
+    CmdHashStart,           // 6
+    CmdHashData,            // 7
+    CmdHashEnd,             // 8
+    CmdCancelTpmCmd,        // 9
+    CmdStoreVolatile,       // 10
+    CmdResetTpmEstablished, // 11
+    CmdGetStateBlob,        // 12
+    CmdSetStateBlob,        // 13
+    CmdStop,                // 14
+    CmdGetConfig,           // 15
+    CmdSetDatafd,           // 16
+    CmdSetBufferSize,       // 17
+}
+
+#[derive(Error, Debug)]
+pub enum TPMIocError {
```

Comment:
There is not need to namespace the enum name Rust gives you namespaces so you can just call it `Error`